src/components/FreeClass/DateTimePicker/DateTimePicker.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { FC, useState } from "react"
+import { Pressable, View } from "react-native"
+import { Text } from "components/Text"
+import { usePalette } from "utils/colors"
+import { DateTimeBox } from "./DateTimeBox"
+import { destructureDate } from "utils/dates"
+import DateTimePickerModal from "react-native-modal-datetime-picker"
+export interface DateTimePickerProps {
+    date: Date
+    setDate: (date: Date) => void
+}
+
+/**
+ * custom DateTime picker
+ */
+export const DateTimePicker: FC<DateTimePickerProps> = props => {
+    const { year, month, day, hour, minute } = destructureDate(props.date)
+    const { isLight, primary } = usePalette()
+
+    const [isDatePickerVisible, setDatePickerVisibility] = useState(false)
+
+    //there would be also "date-time" but we don't use it
+    const [dateMode, setDateMode] = useState<"date" | "time">("date")
+
+    const showDatePicker = () => {
+        setDateMode("date")
+        setDatePickerVisibility(true)
+    }
+
+    const showTimePicker = () => {
+        setDateMode("time")
+        setDatePickerVisibility(true)
+    }
+
+    const hideDateOrTimePicker = () => {
+        setDatePickerVisibility(false)
+    }
+
+    const handleConfirm = (date: Date) => {
+        props.setDate(date)
+        console.log("A date has been picked: ", date)
```

Comment:
diego te e i console log 😭
no srsly tho, cerchiamo di lasciare solo log che possano essere in qualche modo utili per diagnostica in caso ci siano problemi dell'app, e spero per dio che non sia un datepicker a rompere l'app lmao